### ISSUE TITLE
Build failure on Python 3.8/Win64 (while succesful for other Python versions and platforms)

### ISSUE THREAD

**Issue by SylvainCorlay** [microsoft/ptvsd#2126]
The conda-forge package for ptvsd fails to build on windows with Python 3.8. However, the package builds succesfully for other versions of Python, and other platforms.

```
 _pydevd_frame_eval\pydevd_frame_evaluator.c(7520): error C2037: left of 'eval_frame' specifies undefined struct/union '_is'
    _pydevd_frame_eval\pydevd_frame_evaluator.c(7619): error C2037: left of 'eval_frame' specifies undefined struct/union '_is'
```

A full build log is available in PR https://github.com/conda-forge/ptvsd-feedstock/pull/15.

If a ptvsd maintainer could chime into this issue with the feedstock, it would be fantastic! This is a significant blocker for the packaging of several other project

**Comment by int19h** [comment]
As for the workaround, it might be possible to disable building the native code bits in pydevd for Python 3.8+ via some build flags or options - @fabioz should be able to tell.

**Comment by fabioz** [comment]
Actually, microsoft/debugpy#35 was used to track the pre-release of Python 3.9 (which may still need more work until the final 3.9 release) and this version of ptvsd should work with CPython 3.8.

I believe that the issue is that the Python build on Windows is not providing the CPython internal libraries that the debugger relies on (CPython 3.8 broke many things and moved a bunch of includes to `include/internal` -- besides just `include` and `include/cpython` -- and this `include/internal` is needed to compile the debugger dependencies.

You can see how during the generation of the `.c` files, `setup_cython.py` does apply some workarounds to compile properly given that see: https://github.com/microsoft/ptvsd/blob/master/src/ptvsd/_vendored/pydevd/setup_cython.py#L80). 

And later (also in that file), note how `Py_BUILD_CORE_MODULE` must be set to `1` when compiling to be able to access the `include/internal`.

Note that you can pass `--force-cython` to `setup_cython.py` if you want to regenerate the `.c` files instead of just compiling the existing `.c` files in the repo.

So, in short, I don't think there's much to be done in the debugger, it's probably just a matter of fixing the conda recipe (and maybe making sure that the base CPython build being targeted does indeed ship `include/internal`).

**Comment by SylvainCorlay** [comment]
Thanks @fabioz for your response. The cpython build properly ships `include/internal` headers.

However, there is a bit of patching required (in setup.py to force the cythonization, then in setup_cython.py to include the preprocessor flags required to include the internal headers Py_BUILD_CORE_MODULE and Py_BUILD_CORE_BUILTIN).

I wonder how the PyPI wheel of ptvsd 4.3.2 was produced for windows and python 3.8. Was it built with a patched source?

**Comment by SylvainCorlay** [comment]
Applying the patch to the sources in the conda recipe for ptvsd ends up with

` finished with status 'error'` unfortunately.

on, windows + python 3.8 (only).

Should this issue be re-opened?

**Comment by fabioz** [comment]
@SylvainCorlay, sorry for the delay in getting back to you here... 

Unfortunately, `ptvsd` is no longer being supported and no further releases will be done, so, ideally, you'd upgrade to use `debugpy` instead (if you have some error there we should be able to fix it there as even if there's an error in the latest version of ptvsd, no further releases/changes will be done).

As for the 4.3.2 wheel, it was shipped with a standard version of CPython (but it's even possible that newer CPython releases broke what worked at that time -- or maybe it was shipped without binaries -- I honestly don't remember... as a note, some critical issues were found in that optimized the debugger with frame evaluation mode... there's a pull request regarding it to be merged soon on debugpy -- i.e.: https://github.com/microsoft/debugpy/pull/377, so, I'd suggest you don't even ship with the binaries for frame eval with ptvsd).

Note: the cython speed ups are still recommended, just the debugging which uses the CPython frame evaluation mode (which modifies bytecode to add breakpoints without tracing) had some issues.
